src/UI/MainMenuOptions.cpp

```diff
@@ -0,0 +1,370 @@
+// This is an open source non-commercial project. Dear PVS-Studio, please check it.
+// PVS-Studio Static Code Analyzer for C, C++ and C#: http://www.viva64.com
+
+#include "MainMenuOptions.h"
+
+#include "NAS2D/Renderer/Renderer.h"
+
+#include "NAs2D/Utility.h"
+#include "NAS2D/Configuration.h"
+#include "NAS2D/Mixer/Mixer.h"
+#include "NAS2D/StringUtils.h"
+
+#include "../Constants.h"
+
+#include <sstream>
+
+#include <SDL.h>
+#include <SDL_video.h>
+
+
+MainMenuOptions::MainMenuOptions() : Window()
+{
+    text(constants::WINDOW_SYSTEM_TITLE);
+    init();
+}
+
+
+MainMenuOptions::~MainMenuOptions()
+{
+    auto& e = NAS2D::Utility<NAS2D::EventHandler>::get();
+    e.keyDown().disconnect(this, &MainMenuOptions::onKeyDown);
+    btnOk.click().disconnect(this, &MainMenuOptions::btnOkClicked);
+    btnCancel.click().disconnect(this, &MainMenuOptions::btnCancelClicked);
+    btnApply.click().disconnect(this, &MainMenuOptions::btnApplyClicked);
+}
+
+
+void MainMenuOptions::init()
+{
+
+    auto& e = NAS2D::Utility<NAS2D::EventHandler>::get();
+    e.keyDown().connect(this, &MainMenuOptions::onKeyDown);
+
+    size(500, 350);
+
+    const auto border_left_width = 5;
+    const auto border_right_width = 5;
+    const auto border_top_height = 25; //WINDOW_TITLE_HEIGHT + 5
+    const auto border_bottom_height = 5;
+    const auto element_spacing = 5;
+    const auto left_edge = positionX() + border_left_width;
+    const auto right_edge = left_edge + width() - border_right_width * 2.0f;
+    const auto top_edge = positionY() + border_top_height;
+    const auto bottom_edge = top_edge + height() - border_top_height - border_bottom_height * 2.0f;
+    const auto width = right_edge - left_edge;
+    const auto height = bottom_edge - top_edge;
+
+    pnlControls.position(left_edge, top_edge);
+    pnlControls.size(width, height);
+
+    pnlButtons.position(pnlControls.positionX(), bottom_edge - element_spacing);
+    pnlButtons.size(pnlControls.width(), 120);
+
+    lblResolution.text("Resolution");
+    lblResolution.size(0, 0);
+
+    cmbResolution.visible(true);
+    cmbResolution.enabled(true);
+    cmbResolution.size(200, 0);
+    cmbResolution.maxDisplayItems(1);
+
+    cmbResolution.clearSelection();
+    //TODO: add functions to the renderer/DeviceContext to get/set display count and modes.
```

Comment:
This code makes a good case for providing such functionality. Perhaps it can return a `std::vector` of structs. A reasonable addition as a separate PR.

----

I guess the existing code doesn't really try to set video modes currently. I would assume this is desirable for a game, and was left out before due to lack of time to implement or simple oversight.